src/sentry/identity/vsts_extension/provider.py

```diff
@@ -0,0 +1,7 @@
+from __future__ import absolute_import
+
+from sentry.identity.vsts.provider import VSTSIdentityProvider
+
+
+class VstsExtensionIdentityProvider(VSTSIdentityProvider):
```

Comment:
@mnoble Hi there. What does the identity provider do in this case? I'm not sure what it's used for here.

Reply:
The same thing the normal VSTS one does. It's just a new class here, since everything is done via `key`, this one needs to match the Integration's key (`vsts-extension`).

Reply:
@mnoble That's a fine way to do it. I guess when I look at this, it seems strange we couldn't just call the vsts one. Does this affect anything else by using the `vsts-extention` identity? For instance, are the new identities now branded with `vsts-extension` instead of `vsts`?

I don't believe that is what we want there. I could be wrong, but I'm assuming we would just want a vsts identity instead or we need to be able to somehow be able to handle it.  Does it affect the login stuff?  @adhiraj could you weigh in?

Reply:
No, the Identity records are all still marked as `vsts`. The Integration's `provider` is `vsts-extension`, but I changed the endpoint to treat that the same as `vsts`. They won't look any different in the list on the Integrations page.

The problem is that the various pipeline classes use the `key` to dynamically determine which class to use. That coupled with the fact that we _need_ a `VstsExtensionIntegrationProvider` class, since the last step is different than the normal VSTS one, means we _also_ need this class here.

If the pipeline classes used dependency injection, instead of the "lookup by key" logic it does now, it might be easier to reuse the normal `VSTSIdentityProvider`. That seems like more work than is worth right now, though.